clcache_monkey.py

```diff
@@ -0,0 +1,30 @@
+import os
+
+from typing import List
+
+
+def patch_distutils() -> None:
+    from distutils import ccompiler
+    
+    ccompiler_spawn = ccompiler.CCompiler.spawn
+    def msvc_compiler_spawn(self, args: List[str]) -> None:
+        if args[0] != self.cc:
```

Comment:
A major assumption here is that clcache is run on windows (eg. if this is run on linux, clcache will still be used because this doesn't patch msvccompiler, which isn't public).